File: QuickStarts/VB/EventBroker/Src/EventBrokerExtension/PublishesAttribute.vb
'===============================================================================
' Microsoft patterns & practices
' Unity Application Block
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================

Imports System
Imports System.Collections.Generic
Imports System.Text

Namespace SimpleEventBroker
	<AttributeUsage(AttributeTargets.[Event], Inherited := True)> _
	Public Class PublishesAttribute
		Inherits PublishSubscribeAttribute
		Public Sub New(ByVal publishedEventName As String)
			MyBase.New(publishedEventName)
		End Sub
	End Class
End Namespace

File: quickstarts/CS/EventBroker/Src/EventBrokerExtension/PublishesAttribute.cs
// Copyright (c) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.

using System;
using System.Collections.Generic;
using System.Text;

namespace SimpleEventBroker
{
    [AttributeUsage(AttributeTargets.Event, Inherited = true)]
    public class PublishesAttribute : PublishSubscribeAttribute
    {
        public PublishesAttribute(string publishedEventName) : base(publishedEventName)
        {
        }
    }
}
